improver_tests/calibration/reliability_calibration/test_ConstructReliabilityCalibrationTables.py

```diff
@@ -0,0 +1,436 @@
+        assert_array_equal(result.points, expected_cycle_hour)
+        self.assertEqual(result.name(), 'cycle_hour')
+
+
+class Test__define_metadata(Test_Setup):
+
+    """Test the _define_metadata method."""
+
+    def test_metadata_with_complete_inputs(self):
+        """Test the metadata returned is complete and as expected when the
+        forecast cube contains the required metadata to copy."""
+
+        self.forecast_1.attributes['institution'] = 'Kitten Inc'
+        self.expected_attributes['institution'] = 'Kitten Inc'
+
+        diagnostic = self.forecast_1.coord(var_name='threshold').name()
+        result = Plugin._define_metadata(self.forecast_1, diagnostic)
+
+        self.assertIsInstance(result, dict)
+        self.assertEqual(result, self.expected_attributes)
+
+    def test_metadata_with_incomplete_inputs(self):
+        """Test the metadata returned is complete and as expected when the
+        forecast cube does not contain all the required metadata to copy."""
+
+        diagnostic = self.forecast_1.coord(var_name='threshold').name()
+        result = Plugin._define_metadata(self.forecast_1, diagnostic)
+
+        self.assertIsInstance(result, dict)
+        self.assertEqual(result, self.expected_attributes)
+
+
+class Test__create_reliability_table_cube(Test_Setup):
+
+    """Test the _create_reliability_table_cube method."""
+
+    def test_valid_inputs(self):
+        """Test the cube returned has the structure expected."""
+
+        forecast_slice = next(self.forecast_1.slices_over('air_temperature'))
+        result = Plugin()._create_reliability_table_cube(
+            forecast_slice, forecast_slice.coord(var_name='threshold'))
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertSequenceEqual(result.shape, self.expected_table_shape)
+        self.assertEqual(result.name(), "reliability_calibration_table")
+        self.assertEqual(result.attributes, self.expected_attributes)
+
+
+class Test__populate_reliability_bins(Test_Setup):
+
+    """Test the _populate_reliability_bins method."""
+
+    def test_table_values(self):
+        """Test the reliability table returned has the expected values for the
+        given inputs."""
+
+        forecast_slice = next(self.forecast_1.slices_over('air_temperature'))
+        truth_slice = next(self.truth_1.slices_over('air_temperature'))
+        result = Plugin()._populate_reliability_bins(
+            forecast_slice.data, truth_slice.data)
+
+        self.assertSequenceEqual(result.shape, self.expected_table_shape)
+        assert_array_equal(result, self.expected_table)
+
+
+class Test_process(Test_Setup):
+
+    """Test the process method."""
+
+    def test_return_type(self):
+        """Test the process method returns a reliability table cube."""
+
+        result = Plugin().process(self.forecasts, self.truths)
+
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result[0].name(), "reliability_calibration_table")
```

Comment:
You don't need to index result?
It also wouldn't hurt to check the threshold coordinate is as expected as this is the thing that should get promoted to a leading dimension.

Reply:
Checked the threshold coordinate is equivalent to that on the original forecast cubes and checked it is the leading dimension.